Code/Source/svFSI/output.cpp

```diff
@@ -146,31 +146,40 @@ void output_result(Simulation* simulation,  std::array<double,3>& timeP, const i
     tmp = 100.0;
   }
 
+  // Add a warning if the solution to the linear system did not converge.
+  std::string convergence_msg;
   if (eq.FSILS.RI.suc) {
      c1 = "["; 
      c2 = "]";
   } else {
      c1 = "!";  
      c2 = "!";
+     convergence_msg = "  WARNING: The linear system solution has not converged";
   }
 
   // NS     1-2  3.82E1  [ -62 7.92E-4 7.92E-4 3.60E-4]  [   5  -15  23] 
   //                                                      -------------
   auto db_str = std::to_string(static_cast<int>(round(eq.FSILS.RI.dB)));
   auto calld_str = std::to_string(static_cast<int>(round(tmp)));
   sOut += "  " + c1 + std::to_string(eq.FSILS.RI.itr) + " " + db_str + " " + calld_str + c2;
+  sOut += convergence_msg; 
 
   if (com_mod.nEq > 1) {
     logger << sOut << std::endl;
   } else {
     logger << sOut << std::endl;
   }
 
-  // if max number of iterations is reached, throw a warning
-  if (eq.itr == eq.maxItr) {
-    logger << "MAX NUMBER OF NONLINEAR ITERATIONS REACHED. CHECK CONVERGENCE." << std::endl;
+  // Print a warning message if the maximum number of nonlinear iterations has been exceeded.
+  if (eq.itr > eq.maxItr) {
```

Comment:
can eq.itr be greater than eq.maxItr? I think the loop stops at eq.itr=eq.maxItr, I am not sure the message will be printed.

Reply:
@dcodoni `eq.itr` can be larger than `eq.maxItr`, my thinking was that if it was equal than that is ok. But what really matters is the size of the residual. 

Note that there is code in the `picc()` that also checks `eq.itr` and other convergence measures
```
  bool l1 = (eq.itr >= eq.maxItr);
  bool l2 = (r1 <= eq.tol);
  bool l3 = (r1 <= eq.tol*eq.pNorm);
  bool l4 = (eq.itr >= eq.minItr);

  if (l1 || ((l2 || l3) && l4)) {
    eq.ok = true;
   }
```
If `l1` if false then `eq.ok` is set to false and later in the code
```
    if (eq.ok) {
      cEq = cEq + 1;
    }
```
so the current equation is not incremented?

Reply:
I see. If l1 is false then eq.ok is set to false only if l2 and l3 are false. At this point the current equation is not incremented. But if l1 is true then the current equation is always incremented because eq.ok is always true.
I think that as soon as eq.itr becomes equal to eq.maxItr the current equation is incremented. For this reason I thought eq.itr would never go above eq.maxItr.
But maybe I am wrong

Reply:
@dcodoni The tests I ran seemed to do the right thing but I can use `if (eq.itr == eq.maxItr) {` if you think it is better.

Reply:
I've done more testing and thins seems to do the right thing. Let's merge it.

Reply:
Sounds good @ktbolt. And thanks for doing this!